### PR TITLE
feature/polycarbdiet s20

### PR DESCRIPTION
**Pull request by polycarbdiet**
## Description

Hello! 

As someone who has worked with AVR microcontrollers for years and regularly participates in mechanical design and PCB design in their day job, I have decided to design personal keyboards so that they can be tailored to my use cases. But after being in the hobby for over a year, I have noticed that there is a lack of in-stock or quick turn around keyboards. 

After my friends, family, and coworkers found out about my niche (at least to them) hobby, they have all been asking for some kind of custom mechanical keyboard.
If I am designing mechanical keyboards for myself, I might as well share them with everyone that is also interested.

This PR contains a `polycarbdiet` directory for me so that when I have more keyboards to add, they will all be neatly organized. I am aware that this is against the norm, maybe even frowned up. While I am working on taking my functional prototypes of the S20 and bringing it to a point where it is production-ready, I have another keyboard design that I am working on for my daily driver at work. I plan on making that keyboard QMK-compatible as well and sharing it with the world. 

The S20 keyboard that I designed and want to bring to a GB phase and private buys for close friends and family is a 5x4 ortholinear MacroPad/NumberPad. It has the possibility to be used in either configuration depending on the user's preference. I have added an additional layout called `LAYOUT_Enter`. This is my prefered layout that provides a hybrid between a macropad and a numberpad. This layout is commonly supported by many custom keycap sets.

Calling me a Git "noob" would not be incorrect. If there is anything that is not up to standards, please let me know and I will try my best to fix it!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by polycarbdiet** [comment]
Sorry about the changes.

This is where I learn the command `git config core.ignorecase false`.

Changed the files to lower case as it was part of the PR Checklist. When pushed to the remote repository, the cases of the file names were not changed. Changed the file names accordingly.

- [x] Files still compile and flash the PCB
- [x] Verified the cases on the remote repository

File: keyboards/polycarbdiet/s20/s20.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    { K30, K31, K32, K33 }, \
+    { K40, K41, K42, K43 }  \
+}
+
+#define LAYOUT_NumberPad(   \
+    K00, K01, K02, K03,  \
+    K10, K11, K12, K13,  \
+    K20, K21, K22,       \
+    K30, K31, K32, K33,  \
+    K40,      K42        \
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/s20.h:40]:
```suggestion
    K10, K11, K12,       \
    K20, K21, K22, K13,  \
    K30, K31, K32,       \
    K40,    K42,   K33   \
```

File: keyboards/polycarbdiet/s20/s20.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+) { \
+    { K00, K01,   K02, K03   }, \
+    { K10, K11,   K12, K13   }, \
+    { K20, K21,   K22, KC_NO }, \
+    { K30, K31,   K32, K33   }, \
+    { K40, KC_NO, K42, KC_NO }  \
+}
+
+#define LAYOUT_Enter(    \
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/s20.h:49]: We activately avoid uppercase for the layout macros, except for the `LAYOUT` part. 
```suggestion
#define LAYOUT_enter(    \
```

File: keyboards/polycarbdiet/s20/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+LAYOUTS = MacroPad NumberPad Enter
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/rules.mk:23]:
```suggestion
LAYOUTS = ortho_5x4 numpad_5x4
```

File: keyboards/polycarbdiet/s20/info.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                {"x":2, "y":3},
+                {"x":3, "y":3},
+
+                {"x":0, "y":4},
+                {"x":1, "y":4},
+                {"x":2, "y":4},
+                {"x":3, "y":4}
+            ]
+        },
+        "LAYOUT_NumberPad":{
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:36]:
```suggestion
        "LAYOUT_numpad_5x4":{
```

File: keyboards/polycarbdiet/s20/info.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            "layout": [
+                {"x":0, "y":0},
+                {"x":1, "y":0},
+                {"x":2, "y":0},
+                {"x":3, "y":0},
+
+                {"x":0, "y":1},
+                {"x":1, "y":1},
+                {"x":2, "y":1},
+                {"x":3, "y":1, "h":2},
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:46]:
```suggestion
```

File: keyboards/polycarbdiet/s20/info.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+                {"x":0, "y":2},
+                {"x":1, "y":2},
+                {"x":2, "y":2},
+
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:51]:
```suggestion
                {"x":3, "y":1, "h":2},
```

File: keyboards/polycarbdiet/s20/info.json [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                {"x":0, "y":3},
+                {"x":1, "y":3},
+                {"x":2, "y":3},
+                {"x":3, "y":3, "h":2},
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:55]:
```suggestion
```

File: keyboards/polycarbdiet/s20/info.json [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+                {"x":0, "y":4, "w":2},
+                {"x":2, "y":4}
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:58]:
```suggestion
                {"x":2, "y":4},
                {"x":3, "y":3, "h":2}
```

File: keyboards/polycarbdiet/s20/info.json [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            ]
+        },
+        "LAYOUT_Enter": {
```
Comment (drashna) [inline keyboards/polycarbdiet/s20/info.json:61]:
```suggestion
        "LAYOUT_enter": {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by polycarbdiet — COMMENTED** [review]
@drashna: Requested changes were made.

**Comment by polycarbdiet** [comment]
Too many unnecessary commits (actually all of them). Will redo PR.

**Pull request closed without merging**